src/components/results/securityanalysis/security-analysis-result-utils.tsx

```diff
@@ -0,0 +1,371 @@
+                computationStatus: postContingencyResult.status,
+            });
+        }
+
+        if (
+            postContingencyResult.limitViolationsResult.limitViolations.length >
+            0
+        ) {
+            postContingencyResult?.limitViolationsResult?.limitViolations?.forEach(
+                (limitViolation) => {
+                    let contingencies: ResultConstraint[];
+                    if (!mapConstraints.has(limitViolation.subjectId)) {
+                        contingencies = [];
+                        mapConstraints.set(
+                            limitViolation.subjectId,
+                            contingencies
+                        );
+                    } else {
+                        contingencies = mapConstraints.get(
+                            limitViolation.subjectId
+                        );
+                    }
+
+                    contingencies.push({
+                        contingencyId: postContingencyResult.contingency.id,
+                        contingencyEquipmentsIds:
+                            postContingencyResult.contingency.elements.map(
+                                (element) => element.id
+                            ),
+                        computationStatus: postContingencyResult.status,
+                        constraintId: limitViolation.subjectId,
+                        limitType: intl.formatMessage({
+                            id: limitViolation.limitType,
+                        }),
+                        limit: limitViolation.limit,
+                        value: limitViolation.value,
+                        loading: limitViolation.loading,
+                        side: limitViolation.side,
+                        acceptableDuration: limitViolation.acceptableDuration,
+                        limitName: limitViolation.limitName,
+                    });
+                }
+            );
+        }
+    });
+
+    mapConstraints.forEach((contingencies, subjectId) => {
+        rows.push({
+            subjectId: subjectId,
+        });
+
+        contingencies?.forEach((contingency: ResultConstraint) => {
+            rows.push({
+                contingencyId: contingency.contingencyId,
+                contingencyEquipmentsIds: contingency.contingencyEquipmentsIds,
+                computationStatus: contingency.computationStatus,
+                constraintId: contingency.constraintId,
+                limitType: contingency.limitType,
+                limit: contingency.limit,
+                value: contingency.value,
+                loading: contingency.loading,
+                side: contingency.side,
+                acceptableDuration: contingency.acceptableDuration,
+                limitName: contingency.limitName,
+                linkedElementId: subjectId,
+            });
+        });
+    });
+
+    return rows;
+};
+
+export const securityAnalysisTableNmKConstraintsColumnsDefinition = (
```

Comment:
securityAnalysisTableNmKConstraintsColumnsDefinition and securityAnalysisTableNmKContingenciesColumnsDefinition have the same interface

make a specific interface for them

Reply:
Already done.